In [1]:
import os
os.environ['DJANGO_ALLOW_ASYNC_UNSAFE'] = "true"

# import modules
from shop.models import *

1.1 query หาข้อมูล Order ทั้งหมดที่เกิดขึ้นในเดือน เมษายน มาแสดงผล 5 รายการแรก และแสดงผลดังตัวอย่าง 

ORDER ID:1, DATE: 2024-04-24, AMOUNT: 3
ORDER ID:1, DATE: 2024-04-24, AMOUNT: 3
ORDER ID:2, DATE: 2024-04-24, AMOUNT: 1
ORDER ID:2, DATE: 2024-04-24, AMOUNT: 2
ORDER ID:3, DATE: 2024-04-24, AMOUNT: 3


ORDER ID: 1, DATE: 2024-04-24
ORDER ID: 2, DATE: 2024-04-24
ORDER ID: 3, DATE: 2024-04-24
ORDER ID: 4, DATE: 2024-04-25
ORDER ID: 5, DATE: 2024-04-25


1.2 query หาข้อมูล Product ที่มีคำว่า Classic ในรายละเอียดสินค้า และแสดงผลดังตัวอย่าง

PRODUCT ID:25, DESCRIPTION: Classic button-down shirt suitable for formal wear., AMOUNT: 428
PRODUCT ID:44, DESCRIPTION: Classic board game of buying, trading, and strategy., AMOUNT: 90
PRODUCT ID:47, DESCRIPTION: Classic block-stacking game for parties and gatherings., AMOUNT: 30
PRODUCT ID:61, DESCRIPTION: Classic round-cut diamond earrings in 14k white gold., AMOUNT: 10


PRODUCT ID:25, DESCRIPTION:Classic button-down shirt suitable for formal wear.
PRODUCT ID:44, DESCRIPTION:Classic board game of buying, trading, and strategy.
PRODUCT ID:47, DESCRIPTION:Classic block-stacking game for parties and gatherings.
PRODUCT ID:61, DESCRIPTION:Classic round-cut diamond earrings in 14k white gold.


1.3 หาข้อมูล Payment ที่มีราคามากกว่า 500.00 และน้อยกว่า 2000.00  มาแสดงผล 20 รายการแรก และแสดงผลดังตัวอย่าง

PAYMENT ID:62, DATE: 2024-05-13, PRICE: 600.00
PAYMENT ID:3, DATE: 2024-04-24, PRICE: 700.00
PAYMENT ID:5, DATE: 2024-04-24, PRICE: 590.00
PAYMENT ID:9, DATE: 2024-04-26, PRICE: 599.00
PAYMENT ID:17, DATE: 2024-04-28, PRICE: 699.00
PAYMENT ID:20, DATE: 2024-04-29, PRICE: 990.00
PAYMENT ID:22, DATE: 2024-04-30, PRICE: 590.00
PAYMENT ID:26, DATE: 2024-05-01, PRICE: 1720.00
PAYMENT ID:29, DATE: 2024-05-02, PRICE: 1190.00
PAYMENT ID:31, DATE: 2024-05-03, PRICE: 700.00
PAYMENT ID:32, DATE: 2024-05-03, PRICE: 1690.00
PAYMENT ID:34, DATE: 2024-05-04, PRICE: 1140.00
PAYMENT ID:35, DATE: 2024-05-04, PRICE: 1299.00
PAYMENT ID:40, DATE: 2024-05-06, PRICE: 540.00
PAYMENT ID:41, DATE: 2024-05-06, PRICE: 1400.00
PAYMENT ID:48, DATE: 2024-05-09, PRICE: 1040.00
PAYMENT ID:50, DATE: 2024-05-09, PRICE: 1699.00
PAYMENT ID:51, DATE: 2024-05-10, PRICE: 1649.00
PAYMENT ID:53, DATE: 2024-05-10, PRICE: 1400.00
PAYMENT ID:54, DATE: 2024-05-11, PRICE: 540.00


PAYMENT ID:62, DATE: 2024-06-25, PRICE: 600.00
PAYMENT ID:3, DATE: 2024-06-25, PRICE: 700.00
PAYMENT ID:5, DATE: 2024-06-25, PRICE: 590.00
PAYMENT ID:9, DATE: 2024-06-25, PRICE: 599.00
PAYMENT ID:17, DATE: 2024-06-26, PRICE: 699.00
PAYMENT ID:20, DATE: 2024-06-26, PRICE: 990.00
PAYMENT ID:22, DATE: 2024-06-26, PRICE: 590.00
PAYMENT ID:26, DATE: 2024-06-25, PRICE: 1720.00
PAYMENT ID:29, DATE: 2024-06-25, PRICE: 1190.00
PAYMENT ID:31, DATE: 2024-06-27, PRICE: 700.00
PAYMENT ID:32, DATE: 2024-06-27, PRICE: 1690.00
PAYMENT ID:34, DATE: 2024-06-26, PRICE: 1140.00
PAYMENT ID:35, DATE: 2024-06-26, PRICE: 1299.00
PAYMENT ID:40, DATE: 2024-06-25, PRICE: 540.00
PAYMENT ID:41, DATE: 2024-06-25, PRICE: 1400.00
PAYMENT ID:48, DATE: 2024-06-27, PRICE: 1040.00
PAYMENT ID:50, DATE: 2024-06-26, PRICE: 1699.00
PAYMENT ID:51, DATE: 2024-06-25, PRICE: 1649.00
PAYMENT ID:53, DATE: 2024-06-26, PRICE: 1400.00
PAYMENT ID:54, DATE: 2024-06-25, PRICE: 540.00


1.4 ให้เพิ่มสินค้าใหม่จำนวน 3 รายการ 

สินค้าที่ 1
ชื่อ: Physical Programming
หมวดหมู่สินค้า: Books and Media
จำนวนคงเหลือ: 5
รายละเอียด: By Tanakit.
ราคา: 990

สินค้าที่ 2
ชื่อ: Macbook Pro
หมวดหมู่สินค้า: Information Technology และ Electronics
จำนวนคงเหลือ: 100
รายละเอียด: Game-changing graphics performance.
ราคา: 52465

สินค้าที่ 3
ชื่อ: Sofa
หมวดหมู่สินค้า: Furniture
จำนวนคงเหลือ: 50
รายละเอียดซ: Comfort seat.
ราคา: 10990

1.5 แก้ไขชื่อสินค้า จาก Physical Programming เป็น Introduction to Computer System และ แก้ไขจำนวนสินค้า จาก 5 ชิ้นเป็น 150

1.6 แก้ไขชื่อหมวดหมู่สินค้า จาก Clothing and Apparel เป็น Fashion

1.7 ลบสินค้าทุกตัวที่อยู่ในหมวดหมู่ Furniture

(12, {'shop.Product_categories': 11, 'shop.ProductCategory': 1})